docs/coordinates/matchsep.rst

```diff
@@ -207,38 +219,48 @@ You can also find the nearest 3d matches, different from the on-sky
 separation shown above only when the coordinates were initialized with
 a ``distance``::
 
-    >>> c = SkyCoord(ra=ra1*u.degree, dec=dec1*u.degree, distance=distance1*u.kpc)  # doctest: +SKIP
-    >>> catalog = SkyCoord(ra=ra2*u.degree, dec=dec2*u.degree, distance=distance2*u.kpc)  # doctest: +SKIP
-    >>> idx, d2d, d3d = c.match_to_catalog_3d(catalog)  # doctest: +SKIP
+.. doctest-requires:: scipy
+
+    >>> c = SkyCoord(ra=ra1*u.degree, dec=dec1*u.degree, distance=distance1*u.kpc)
+    >>> catalog = SkyCoord(ra=ra2*u.degree, dec=dec2*u.degree, distance=distance2*u.kpc)
+    >>> idx, d2d, d3d = c.match_to_catalog_3d(catalog)
 
 Now ``idx`` are indices into ``catalog`` that are the closest objects to each
 of the coordinates in ``c``, ``d2d`` are the on-sky distances between them, and
 ``d3d`` are the 3-dimensional distances.  Because coordinate objects support
 indexing, ``idx`` enables easy access to the matched set of coordinates in
 the catalog::
 
-    >>> matches = catalog[idx]  # doctest: +SKIP
-    >>> (matches.separation_3d(c) == d3d).all()  # doctest: +SKIP
+
+.. doctest-requires:: scipy
+
+    >>> matches = catalog[idx]
+    >>> (matches.separation_3d(c) == d3d).all()
     True
-    >>> dra, ddec = c.spherical_offsets_to(matches)  # doctest: +SKIP
+    >>> dra, ddec = c.spherical_offsets_to(matches)
 
 This functionality can also be accessed from the
 :func:`~astropy.coordinates.match_coordinates_sky` and
 :func:`~astropy.coordinates.match_coordinates_3d` functions. These
 will work on either |skycoord| objects *or* the lower-level frame classes::
 
+
+.. doctest-requires:: scipy
+
     >>> from astropy.coordinates import match_coordinates_sky
-    >>> idx, d2d, d3d = match_coordinates_sky(c, catalog)  # doctest: +SKIP
-    >>> idx, d2d, d3d = match_coordinates_sky(c.frame, catalog.frame)  # doctest: +SKIP
+    >>> idx, d2d, d3d = match_coordinates_sky(c, catalog)
+    >>> idx, d2d, d3d = match_coordinates_sky(c.frame, catalog.frame)
 
 It is possible to impose a separation constraint (e.g., the maximum separation to be
 considered a match) by creating a boolean mask with ``d2d`` or ``d3d``. For example,::
 
-    >>> max_sep = 1.0 * u.arcsec # doctest: +SKIP
-    >>> idx, d2d, d3d = c.match_to_catalog_3d(catalog) # doctest: +SKIP
-    >>> sep_constraint = idx < max_sep # doctest: +SKIP
-    >>> c_matches = c[sep_constraint] # doctest: +SKIP
-    >>> catalog_matches = c[idx[sep_constraint]] # doctest: +SKIP
+.. doctest-requires:: scipy
+
+    >>> max_sep = 1.0 * u.arcsec
+    >>> idx, d2d, d3d = c.match_to_catalog_3d(catalog)
+    >>> sep_constraint = idx < max_sep
+    >>> c_matches = c[sep_constraint]
+    >>> catalog_matches = c[idx[sep_constraint]]
```

Comment:
Thanks for enabling the doctest and making it pass again. However, there is still something missing as these tests did not catch the typo that multiple people have reported in #8491 and #8531. 🤔

Reply:
@pllim - I think the the additions around https://github.com/astropy/astropy/pull/8412/files#diff-358fb678c45159516eb340b9ed00fe86R198 are dealing with that issue, however you're right that even with this fix the examples are not copy pasteable as is.